boa_interner/src/lib.rs

```diff
@@ -251,13 +261,15 @@ impl Interner {
     /// Gets the symbol of the common string if one of them
     fn get_common(string: &str) -> Option<Sym> {
         COMMON_STRINGS.get_index(string).map(|idx|
-            // SAFETY: `idx >= 0`, since it's an `usize`,
-            // and `idx + 1 > 0` (considering no overflows, but
-            // an overflow would panic on debug anyways).
+            // SAFETY: `idx >= 0`, since it's an `usize`, and `idx + 1 > 0`.
+            // In this case, we don't need to worry about overflows
+            // because `COMMON_STRINGS` would need to be of considerable
+            // size to cause an overflow, even on machines with `usize = u32`.
```

Comment:
Hmmm, I still don't like this approach. To me, it looks like "since we would need many common strings to be unsafe, and we don't have so many, this is safe", but what if this grows in the future? What about 16-bit or 8-bit platforms?

Those probably have other issues of their own, but the only situation where we can ensure that this `unsafe` call is actually safe is if we have really checked the invariants beforehand, and panic if they are not filled, even if this would add some performance impact.

It's similar to the bounds checking in the standard library.

I would suggest to do some compile-time check or an assert on the length of `COMMON_STRINGS` to make sure that the amount of common strings is < `usize::MAX`, and maybe add a comment in `COMMON_STRINGS` itself, to clarify that it cannot have that value.

The thing is that while I guess having a list of length `usize::MAX + 1` would fail to compile, having it the size of `usize::MAX` would actually work, but fail here. But since that list is generated in compile time, and we know the size of `usize::MAX` at compile time, we can compare those then, and not introduce the burden in runtime. Anyways, I know it's a very small edge case, so if it's to complex to generate a `compile_error!()` or the `assert!()` at the beginning of the function doesn't get optimized away (even if `len()` is `const`), I'm OK with leaving it like this.

Reply:
Had to include the `static_assertions` crate for this, but now we have a compile time check in place that verifies `COMMON_STRINGS.len() < usize::MAX` :)

Reply:
Great!! Thank you!!!